client/src/includes/a11y-result.ts

```diff
@@ -0,0 +1,136 @@
+): WagtailAxeConfiguration | null => {
+  const script = container?.querySelector<HTMLScriptElement>(
+    '#accessibility-axe-configuration',
+  );
+
+  if (!script || !script.textContent) return null;
+
+  try {
+    return JSON.parse(script.textContent);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error('Error loading Axe config');
+    // eslint-disable-next-line no-console
+    console.error(err);
+  }
+
+  // Skip initialization of Axe if config fails to load
+  return null;
+};
+
+/**
+ * Render A11Y results based on template elements.
+ */
+export const renderA11YResults = (
+  container: HTMLElement,
+  results: AxeResults,
+  config: WagtailAxeConfiguration,
+  a11yRowTemplate: HTMLTemplateElement,
+  a11ySelectorTemplate: HTMLTemplateElement,
+  onClickSelector: (selectorName: string, event: MouseEvent) => void,
+) => {
+  // Reset contents to support multiple runs of Axe checks in the preview panel
+  // eslint-disable-next-line no-param-reassign
+  container.innerHTML = '';
+
+  if (results.violations.length) {
+    const sortedViolations = sortAxeViolations(results.violations);
+    sortedViolations.forEach((violation, violationIndex) => {
+      container.appendChild(a11yRowTemplate.content.cloneNode(true));
+      const currentA11yRow = container.querySelectorAll<HTMLDivElement>(
+        '[data-a11y-result-row]',
+      )[violationIndex];
+
+      const a11yErrorName = currentA11yRow.querySelector(
+        '[data-a11y-result-name]',
+      ) as HTMLSpanElement;
+      a11yErrorName.id = `w-a11y-result__name-${violationIndex}`;
+      // Display custom error messages supplied by Wagtail if available,
+      // fallback to default error message from Axe
+      a11yErrorName.textContent =
+        config.messages[violation.id] || violation.help;
+      const a11yErrorCount = currentA11yRow.querySelector(
+        '[data-a11y-result-count]',
+      ) as HTMLSpanElement;
+      a11yErrorCount.textContent = `${violation.nodes.length}`;
+
+      const a11yErrorContainer = currentA11yRow.querySelector(
+        '[data-a11y-result-container]',
+      ) as HTMLDivElement;
+
+      violation.nodes.forEach((node, nodeIndex) => {
+        a11yErrorContainer.appendChild(
+          a11ySelectorTemplate.content.cloneNode(true),
+        );
+        const currentA11ySelector =
+          a11yErrorContainer.querySelectorAll<HTMLButtonElement>(
+            '[data-a11y-result-selector]',
+          )[nodeIndex];
+
+        currentA11ySelector.setAttribute('aria-describedby', a11yErrorName.id);
+        const currentA11ySelectorText = currentA11ySelector.querySelector(
+          '[data-a11y-result-selector-text]',
+        ) as HTMLSpanElement;
+        // Special-case when displaying accessibility results within the admin interface.
+        const selectorName =
+          node.target[0] === '#preview-iframe'
+            ? node.target[1]
+            : node.target[0];
```

Comment:
Only difference in how this function works vs. before – special-case so we show Axe selectors as if they were inside the preview iframe.